main.py

```diff
@@ -72,6 +72,8 @@ def update_colors(self, links):
         with self._switches_lock:
             for switch in self.controller.switches.copy().values():
                 if not switch.is_enabled():
+                    if switch.dpid in self.switches:
+                        self.switches[switch.dpid]['neighbors'] = set()
```

Comment:
Even though a disabled switch should not be taken into account in switches, its field `neighbors` needed to be updated to be empty. With the updated done in `topology` PR, it is not possible to have an enabled interface and/or link when a switch is disabled. Therefore, a disabled switch does not have neighbors.
Updated unit test too.